website/public/js/controllers/inventoryCtrl.js

```diff
@@ -112,6 +115,10 @@ habitrpg.controller("InventoryCtrl",
       var potName = Content.hatchingPotions[potion.key].text();
       if (!$window.confirm(window.env.t('hatchAPot', {potion: potName, egg: eggName}))) return;
       user.ops.hatch({params:{egg:egg.key, hatchingPotion:potion.key}});
+      if (!user.preferences.suppressModals.hatchPet) {
+        $rootScope.hatchedPet = {egg: eggName, potion: potName, eggKey: egg.key, pet: 'Pet-' + egg.key + '-' + potion.key};
```

Comment:
Yeah, you [inject it](http://stackoverflow.com/a/26747669/2601552) using [the resolve option](https://github.com/HabitRPG/habitrpg/blob/develop/website/public/js/controllers/rootCtrl.js#L135)